osc-server/src/test/java/org/osc/core/broker/service/tasks/conformance/openstack/securitygroup/UpdateOrDeleteK8sSecurityGroupMetaTaskTest.java

```diff
@@ -0,0 +1,134 @@
+import org.osc.core.test.util.TaskGraphHelper;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.modules.junit4.PowerMockRunnerDelegate;
+
+@RunWith(PowerMockRunner.class)
+@PowerMockRunnerDelegate(value = Parameterized.class)
+public class UpdateOrDeleteK8sSecurityGroupMetaTaskTest {
+    public EntityManager em;
+
+    private SecurityGroup sg;
+
+    private TaskGraph expectedGraph;
+
+    private UpdateOrDeleteK8sSecurityGroupMetaTaskTestData testData;
+
+    public UpdateOrDeleteK8sSecurityGroupMetaTaskTest(
+                    UpdateOrDeleteK8sSecurityGroupMetaTaskTestData testData,
+                    SecurityGroup sg,
+                    TaskGraph expectedGraph) {
+        this.testData = testData;
+        this.sg = sg;
+        this.expectedGraph = expectedGraph;
+    }
+
+    @Before
+    public void testInitialize() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        this.em = InMemDB.getEntityManagerFactory().createEntityManager();
+        populateDatabase();
+        this.expectedGraph = this.testData.createNoMcPolicyMappingK8sGraph(this.sg);
+    }
+
+    private void populateDatabase() {
+        this.em.getTransaction().begin();
+
+        txPersistVS(this.testData.NO_MC_POLICY_MAPPING_SUPPORTED_LABELS_1_VS);
+        txPersistVS(this.testData.NO_MC_POLICY_MAPPING_SUPPORTED_LABELS_4_VS);
+
+
+        this.em.persist(this.sg);
+        this.sg.getSecurityGroupMembers().forEach(sgm -> {
+            this.em.persist(sgm.getLabel());
+            this.em.persist(sgm);
+        });
+
+        this.sg = this.em.find(SecurityGroup.class, this.sg.getId());
+        this.em.getTransaction().commit();
+        this.sg.getSecurityGroupMembers();
+    }
+
+    private void txPersistVS(VirtualSystem vs) {
+        this.em.persist(vs.getVirtualizationConnector());
+        this.em.persist(vs.getDomain().getApplianceManagerConnector());
+        this.em.persist(vs.getDomain());
+        this.em.persist(vs.getApplianceSoftwareVersion().getAppliance());
+        this.em.persist(vs.getApplianceSoftwareVersion());
+        this.em.persist(vs.getDistributedAppliance());
+        this.em.persist(vs);
+    }
+
+    @After
+    public void testTearDown() {
+        InMemDB.shutdown();
+    }
+
+    @Test
+    public void testExecuteTransaction_WithVariousDeploymentSpecs_ExpectsCorrectTaskGraph() throws Exception {
+        // Arrange.
+        UpdateOrDeleteK8sSecurityGroupMetaTask task = new UpdateOrDeleteK8sSecurityGroupMetaTask();
+        task.checkK8sSecurityGroupLabelMetaTask = new CheckK8sSecurityGroupLabelMetaTask();
+        task = task.create(this.sg);
+
+        // Act.
+        task.executeTransaction(this.em);
+
+        // Assert.
+        TaskGraphHelper.validateTaskGraph(task, this.expectedGraph);
+    }
+
+    @Parameters()
+    public static Collection<Object[]> getTestData() {
+        // These must be separate instances because otherwise the
```

Comment:
We have not needed to do this approach yet for any existing tests. Not sure why this is needed here, we can discuss it offline.